File: solutions/3146. Permutation Difference between Two Strings/3146.cpp
// Time complexity: O(n)
// Space complexity: O(1)
class Solution {
 public:
  int findPermutationDifference(string_view s, string_view t) {
    int ans = 0;
    int indices[26] = {};

    for (int i = 0; i < t.length(); ++i)
      indices[t[i] - 'a'] = i;

    for (int i = 0; i < s.length(); ++i)
      ans += abs(i - indices[s[i] - 'a']);

    return ans;    
  }
};

File: solutions/3146. Permutation Difference between Two Strings/3146.py
# Time complexity: O(n)
# Space complexity: O(1)
class Solution:
  def findPermutationDifference(self, s: str, t: str) -> int:
    ans = 0
    indices = [0] * 26

    for i, ch in enumerate(t):
      indices[ord(ch) - ord('a')] = i

    for i, ch in enumerate(s):
      ans += abs(i - indices[ord(ch) - ord('a')])

    return ans